<?xml version="1.0"?>
<launch>
    <arg name="result_topic" default="/mas_perception/detection_result"/>
    <!-- image source -->
    <arg name="image_directory" default=""/>
    <arg name="cloud_topic" default=""/>
    <arg name="image_topic" default=""/>
    <!-- detection -->
    <arg name="class_annotations" default="$(find mas_perception_libs)/models/class_annotation_example.yml" />
    <arg name="kwargs_file" default="$(find mas_perception_libs)/models/image_detector_test_kwargs.yml" />
    <arg name="detection_module" default="mas_perception_libs" />
    <arg name="detection_class" default="ImageDetectorTest" />

    <group ns="mas_perception">
        <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
            <remap from="image" to="$(arg result_topic)" />
            <param name="autosize" type="bool" value="True" />
        </node>

        <node pkg="mas_perception_libs" type="image_detection_test" name="image_detection_client" output="screen">
            <param name="result_topic" type="string" value="$(arg result_topic)" />
            <param name="image_directory" type="string" value="$(arg image_directory)" />
            <param name="cloud_topic" type="string" value="$(arg cloud_topic)" />
            <param name="image_topic" type="string" value="$(arg image_topic)" />
            <param name="class_annotations" type="string" value="$(arg class_annotations)" />
            <param name="kwargs_file" type="string" value="$(arg kwargs_file)" />
            <param name="detection_module" type="string" value="$(arg detection_module)" />
            <param name="detection_class" type="string" value="$(arg detection_class)" />
        </node>
    </group>
</launch>
